### DISCUSSION TITLE
Failed to build theia web app with 1.1.0-RC04

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#873] [Q&A] [answered]
I try to upgrade my project from GLSP 1.1.0-RC03 to 1.1.0-RC04. I upgarted both - `@eclipse-glsp/theia-integration: "1.1.0-RC04` and `@eclipse-glsp/client": "1.1.0-RC04`

After some refactoring and minor changes regarding changes like the switch from `createClientContainer` to `createDiagramContainer`, compiling my modules works fine, instead of the app module. I got a lot (16) errors like this one which I can't resolve:

```
@open-bpmn/open-bpmn-app: ERROR in ../node_modules/@eclipse-glsp/theia-integration/lib/browser/copy-paste-context-menu-contribution.js 29:24-48
@open-bpmn/open-bpmn-app: Module not found: Error: Can't resolve 'sprotty-theia' in '/home/rsoika/git/open-bpmn/open-bpmn.glsp-client/node_modules/@eclipse-glsp/theia-integration/lib/browser'
@open-bpmn/open-bpmn-app:  @ ../node_modules/@eclipse-glsp/theia-integration/lib/browser/index.js 33:13-62
@open-bpmn/open-bpmn-app:  @ ../open-bpmn-theia/lib/browser/frontend-module.js 19:28-70
@open-bpmn/open-bpmn-app:  @ ./src-gen/frontend/index.js 63:31-95
```

Can someone give me a hint from where these kind of errors are coming from?
Is there a branch of the workflow-example available where I can peek how RC04 should be configured? I guess I have now again some dependency mismatches.

**Comment by tortmayr** [comment, marked as answer]
Hi @rsoika.
we [recently](https://github.com/eclipse-glsp/glsp-theia-integration/pull/138) switched all theia extension dependencies to peer dependencies. Overall this makes dependency resolution more stable and should eliminate the need for manual `resolution` blocks.
However, this also means you have to manually declare the version of sprotty-theia that you want to use in your browser-app e.g:

https://github.com/eclipse-glsp/glsp-theia-integration/blob/1889d42d5a31e411d5bbcdd6d35c53b195c38a42/examples/browser-app/package.json#L25

**Reply by rsoika** [reply]
Thanks. I was not aware of this and thought my existing perDependencies, which I have had in my root package.json, was only temporarily and so I removed it total. This was a fault.
Now I added the depedency into my browser-app module and it is building again!
I think this will also solve my problem #872 with the vscode-integration. 

I will now align my project to the workflow example in glsp-thea-integration:V1.1.0-RC04 and will remove the still existing resolutions and perDependencies....

**Reply by rsoika** [reply]
Hi @tortmayr,
puh! This was a tough ride, but finally I managed the update my code base to GLSP and Theia-Integration with version 1.1.0-RC04.  
My existing theia-integration was really very different and I'm surprised that I got this far at all. 

Now I'm very happy that I found back to your current development stream :-)
